File: LearnDemo/app/src/main/java/com/example/learndemo/MyViewModel.java
package com.example.learndemo;

import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

public class MyViewModel extends ViewModel {
    private MutableLiveData<Integer> num;

    public MutableLiveData<Integer> getNum() {
        if (num == null){
            num = new MutableLiveData<>();
            num.setValue(0 );
        }
        return num;
    }

    public void setNum(MutableLiveData<Integer> num) {
        this.num = num;
    }

    public void add(){
        num.setValue(num.getValue()+1);
    }
}

File: JetPackOfKotlin/app/src/main/java/com/example/jetpackofkotlin/MyViewModel/MyViewModel.kt
package com.example.jetpackofkotlin.MyViewModel

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

class MyViewModel: ViewModel() {
    private val _num:MutableLiveData<Int> by lazy { MutableLiveData<Int>().also { it.value = 0 } }

    val num: LiveData<Int>
        get() = _num
        
    fun modifyNum(number:Int){
        _num.value = _num.value?.plus(number)
    }
    /**
     * 初始化 way1
     * way2 lazy...
     */
//    init {
//        num = MutableLiveData()
//        num.value = 0
//    }

}